src/openssl/lib_crypto.cr

```diff
@@ -1,5 +1,15 @@
-@[Link("crypto")]
+ifdef darwin
+  @[Link("crypto")]
+else
+  @[Link(ldflags: "`pkg-config --libs libcrypto`")]
+end
 lib LibCrypto
+  ifdef darwin
+    OPENSSL_102 = false
+  else
+    OPENSSL_102 = {{`(pkg-config --atleast-version=1.0.2 libcrypto && echo true) || echo false`.id}}
```

Comment:
Well, that would have been too easy I guess. In the latter `{% if`s this is still a `MacroExpression`. @asterite how do I run something at compile time once and use its result in different places at compile time without running it again and again?